module-system/utils/sov-data-generators/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+use std::rc::Rc;
+
+use borsh::ser::BorshSerialize;
+use sov_modules_api::default_context::DefaultContext;
+use sov_modules_api::default_signature::private_key::DefaultPrivateKey;
+use sov_modules_api::transaction::Transaction;
+pub use sov_modules_api::EncodeCall;
+use sov_modules_api::{Address, Context, Module};
+use sov_modules_stf_template::{Batch, RawTx, SequencerOutcome, TxEffect};
+use sov_rollup_interface::mocks::TestBlob;
+use sov_rollup_interface::stf::BatchReceipt;
+pub mod bank_data;
+pub mod election_data;
+pub mod value_setter_data;
+
+pub fn new_test_blob_from_batch(batch: Batch, address: &[u8], hash: [u8; 32]) -> TestBlob<Address> {
+    let address = Address::try_from(address).unwrap();
+    let data = batch.try_to_vec().unwrap();
+    TestBlob::new(data, address, hash)
+}
+
+pub fn has_tx_events(apply_blob_outcome: &BatchReceipt<SequencerOutcome, TxEffect>) -> bool {
+    let events = apply_blob_outcome
+        .tx_receipts
+        .iter()
+        .flat_map(|receipts| receipts.events.iter());
+
+    events.peekable().peek().is_some()
+}
+
+pub trait MessageGenerator {
+    type Module: Module;
+
+    fn create_messages(
```

Comment:
Could you please document what is returned? I suppose 3rd element is nonce, but better to clear it out

Reply:
Another approach would be to create a struct that has each field named and documented and return vector of those structs.